### PR TITLE
Resolved error!

### PR DESCRIPTION
**Pull request by AndrejSafundzic**

### CODE REVIEW

**Comment by AndrejSafundzic** [comment]
1. Added that node cannot connect to itself
2. Resolved error:
The problem was that the message queue was empty. You cannot send anything that is empty. "It is only possible to either send strings or byte arrays through ZeroMq, out-of-the-box. Anything else needs to be serialized before passing it to ZeroMq for transfer on the wire.
You can use whatever you like to serialize it, for example Protocol Buffers, JSON or Message Pack. Note that any recipients needs to be able to deserialize the data using the same protocol."
https://stackoverflow.com/questions/14797806/sending-an-existing-dict-through-zmq-ipc

File: src/processes.py [hunk 1]
```diff
@@ -58,15 +58,20 @@ def __init__(self, node: BaseNode, cluster: Cluster, flags=defaultdict(lambda: F
         # Inspired from https://github.com/streed/simpleRaft
         class SubscribeThread(Thread):
             def run(thread):
+                print("FIrst" + thread.name)
                 log.debug('%s starting subscriber thread', self.node.get_name())
-
+                print("Second " + thread.name)
```
Comment (raokrutarth) [inline src/processes.py:63]: please use log.debug or log.info

Comment (AndrejSafundzic) [reply src/processes.py:63]: I'll delete it! in the next merge

Comment (AndrejSafundzic) [reply src/processes.py]: done!

**Pull request merged**